lms/djangoapps/courseware/views/views.py

```diff
@@ -737,6 +737,13 @@ def _progress(request, course_key, student_id):
         and certs_api.cert_generation_enabled(course_key)
     )
 
+    # If current certificate is invalidated by instructor
+    # then don't show the generate button.
+    is_invalid = False
+    if show_generate_cert_btn and certs_api.is_certificate_invalid(student, course_key):
```

Comment:
Instead of toggle the `show_generate_cert_btn` flag here, why can't you incorporate the value of `certs_api.is_certificate_invalid` into the flag definition on line 735?